src/models/room.ts

```diff
@@ -2254,6 +2311,152 @@ export class Room extends EventEmitter {
             return "Empty room";
         }
     }
+
+    /**
+     * When we receive a new visibility change event:
+     *
+     * - store this visibility change alongside the timeline, in case we
+     *   later need to apply it to an event that we haven't received yet;
+     * - if we have already received the event whose visibility has changed,
+     *   patch it to reflect the visibility change and inform listeners.
+     */
+    private applyNewVisibilityEvent(event: MatrixEvent): void {
+        const visibilityChange = event.asVisibilityChange();
+        if (!visibilityChange) {
+            // The event is ill-formed.
+            return;
+        }
+
+        // Ignore visibility change events that are not emitted by moderators.
+        const userId = event.getSender();
+        if (!userId) {
+            return;
+        }
+        if (!EVENT_VISIBILITY_CHANGE_TYPE.some(name => this.currentState.maySendStateEvent(name, userId))) {
```

Comment:
I'm not really convinced that we need an iterator on this - we can just call `maySendStateEvent` twice. It'd be a bit different if we were supporting more than two namespaces or the classes had a purpose outside of detecting presence within an object.

Or in other words: the usage of `some` feels backwards here. A namespaced value is intended to retrieve values from an object/array, not process an entity to check compliance. 

Storing two variables and checking either is more than fine, I think. The power level check uses local cache so should have no performance impact.

Reply:
It strikes me as error-prone, but I'll stop insisting.